compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -628,7 +628,10 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         }
 
         self.infcx.probe(|_snapshot| {
-            if obligation.has_non_region_late_bound() {
+            // FIXME(non_lifetime_binders): Really, we care only to check that we don't
+            // have any non-region *escaping* bound vars, but that's hard to check and
+            // we shouldn't really ever encounter those anyways.
+            if obligation.self_ty().has_non_region_late_bound() {
```

Comment:
I think this is reachable in the code below. If so, please either remove the comment or update it to refer a new issue or to a known-bug ui test.
```rust
// known-bug
#![feature(non_lifetime_binders)]

fn trivial<A>()
where
    for<'a> dyn Fn(A, &'a u8): Fn(A, &'a u8),
{
}

fn main() {
    trivial::<u8>(); //~ ERROR
}
```

Reply:
*updated the example

Reply:
Sorry, I should've been more clear in this comment. I will add a known-bug and update the comment to say it's not reachable **without `feature(non_lifetime_binders)`**.

Reply:
Wait, why don't I just fix this the right way? 😆